services/httpd/requests.go

```diff
@@ -0,0 +1,113 @@
+package httpd
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/influxdata/influxdb/services/meta"
+)
+
+type RequestInfo struct {
+	IPAddr   string
+	Username string
+}
+
+func (r *RequestInfo) String() string {
+	if r.Username != "" {
+		return fmt.Sprintf("%s:%s", r.Username, r.IPAddr)
+	}
+	return r.IPAddr
+}
+
+type RequestTracker struct {
+	requests map[RequestInfo]*int64
+	mu       sync.RWMutex
+	wg       sync.WaitGroup
+	done     chan struct{}
+}
+
+func NewRequestTracker() *RequestTracker {
+	rt := &RequestTracker{
+		requests: make(map[RequestInfo]*int64),
+		done:     make(chan struct{}),
+	}
+	rt.wg.Add(1)
+	go rt.loop()
+	return rt
+}
+
+func (rt *RequestTracker) Increment(req *http.Request, user *meta.UserInfo) {
+	var info RequestInfo
+	host, _, err := net.SplitHostPort(req.RemoteAddr)
+	if err != nil {
+		return
+	}
+
+	info.IPAddr = host
+	if user != nil {
+		info.Username = user.Name
+	}
+
+	// There is an entry in the request tracker. Increment that.
+	if n, ok := rt.requests[info]; ok {
```

Comment:
Concurrent readers are fine provided there are no writers.  There is a writer here: https://github.com/influxdata/influxdb/pull/8350/files#diff-badb764b9e9789c5a3bf719f6c815aadR71

There is also a data race because the map is replaced here: https://github.com/influxdata/influxdb/pull/8350/files#diff-badb764b9e9789c5a3bf719f6c815aadR110